src/lib/wildcard.ts

```diff
@@ -0,0 +1,115 @@
+import fs from 'fs/promises'
+import type { Dirent } from 'fs'
+import path from 'path'
+import { SRC } from '../constants'
+import { ExportCondition } from '../types'
+import {
+  filenameWithoutExtension,
+  hasAvailableExtension,
+  nonNullable,
+} from '../utils'
+
+// TODO: support nested wildcard exportsCondition (e.g. './foo/*')
+const getWildcardExports = (
+  exportsCondition: Record<string, ExportCondition>,
+): Record<string, ExportCondition> => {
+  return { './*': exportsCondition['./*'] }
+}
+
+const isExportable = async (
+  dirent: Dirent,
+  pathname: string,
+): Promise<boolean> => {
+  if (dirent.isDirectory()) {
+    const innerDirents = await fs.readdir(path.join(pathname, dirent.name), {
+      withFileTypes: true,
+    })
+    return innerDirents.some(
+      ({ name }) => name.startsWith('index') && hasAvailableExtension(name),
+    )
+  }
+  return (
+    dirent.isFile() &&
+    !dirent.name.startsWith('index') &&
+    hasAvailableExtension(dirent.name)
+  )
+}
+
+async function getExportables(
+  cwd: string,
+  excludeKeys: string[],
+): Promise<string[]> {
+  const pathname = path.resolve(cwd, SRC)
+  const dirents = await fs.readdir(pathname, { withFileTypes: true })
+  const exportables: (string | undefined)[] = await Promise.all(
+    dirents.map(async (dirent) =>
+      (await isExportable(dirent, pathname)) &&
+      !excludeKeys.includes(dirent.name)
+        ? dirent.name
+        : undefined,
+    ),
+  )
+  return exportables.filter(nonNullable)
+}
+
+function mapWildcard(
+  wildcardExports: Record<string, ExportCondition>,
+  exportables: string[],
+): ExportCondition[] {
+  return exportables.map((exportable) => {
+    const filename = exportable.includes('.')
+      ? filenameWithoutExtension(exportable)
+      : undefined
+
+    if (!filename) {
+      return {
+        [`./${exportable}`]: JSON.parse(
```

Comment:
`wildcardExports` is already an export condition map, could we use object manipulation to modify it instead of using string replacing and json serialize/deserialize?